File: python/08_045.py
def min_processing_power(tasks, max_seconds):
  def can_complete(power):
     total_time = 0
     for task in tasks:
        # Ceiling division
        total_time += (task + power - 1) // power
     return total_time <= max_seconds
  
  left, right = 1, max(tasks)
  
  while left < right:
     mid = (left + right) // 2
     if can_complete(mid):
        right = mid
     else:
        left = mid + 1
        
  return left

File: cpp/08_045.cpp
class Solution {
public:
    int minProcessingPower(vector<int>& tasks, 
                          int maxSeconds) {
        int maxTask = *max_element(tasks.begin(), 
                                  tasks.end());
        
        int left = 1;
        int right = maxTask;
        
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (canComplete(tasks, mid, maxSeconds)) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }
        
        return left;
    }
    
private:
    bool canComplete(const vector<int>& tasks, 
                    int power, int maxSeconds) {
        int totalTime = 0;
        for (int task : tasks) {
            // Ceiling division in C++
            totalTime += (task + power - 1) / power;
        }
        return totalTime <= maxSeconds;
    }
};